pkg/workspace/creds.go

```diff
@@ -3,16 +3,45 @@ package workspace
 import (
 	"os"
 
+	"github.com/pulumi/pulumi/sdk/v3/go/common/env"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
 )
 
+// GetCurrentCloudURL returns the URL of the cloud we are currently connected to. This may be empty if we
+// have not logged in. Note if PULUMI_BACKEND_URL is set, the corresponding value is returned
+// instead irrespective of the backend for current project or stored credentials.
+func GetCurrentCloudURL(ws Context, e env.Env, project *workspace.Project) (string, error) {
```

Comment:
Maybe not in this PR, but it might make sense to have this take a `backendURL string` argument.

Reply:
instead of the envvar?